cfme/tests/automate/custom_button/test_infra_objects.py

```diff
@@ -339,3 +380,57 @@ def test_custom_button_expression(appliance, request, setup_obj, button_group, e
             assert not custom_button_group.is_displayed
             setup_obj.add_tag(tag)
             assert custom_button_group.is_displayed
+
+
+@pytest.mark.uncollectif(lambda button_group: "VM_INSTANCE" not in button_group)
+def test_open_url(request, setup_obj, button_group, method):
+    """ Test Open url functionality of custom button.
+
+    Polarion:
+        assignee: ndhandre
+        initialEstimate: 1/2
+        caseimportance: high
+        testSteps:
+            prerequisites:
+                * Appliance with Infra provider
+
+            Steps:
+                * Create ruby method for url functionality
+                * Create custom button group with the Object type
+                * Create a custom button with open_url option and respective method
+                * Navigate to object Detail page
+                * Execute custom button
+                * Check new tab open or not with respective url
+    """
+
+    group, obj_type = button_group
+    button = group.buttons.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        open_url=True,
+        display_for="Single entity",
+        system="Request",
+        request=method.name,
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+
+    # TODO: Move windows handling functionality to browser
+    initial_count = len(view.browser.selenium.window_handles)
+    main_window = view.browser.selenium.current_window_handle
+    custom_button_group.item_select(button.text)
+
+    wait_for(
+        lambda: len(view.browser.selenium.window_handles) > initial_count,
+        timeout=120,
+        message="Check for window open",
+    )
+    open_url_window = set(view.browser.selenium.window_handles) ^ {main_window}
+
+    view.browser.selenium.switch_to_window(open_url_window.pop())
+    assert "example.com" in view.browser.url
+    view.browser.selenium.close()
```

Comment:
These need to be in a finalizer, in case the assertion fails. Can't leave the browser on the wrong tab.